### DISCUSSION TITLE
bind could take an array of state

### DISCUSSION THREAD

**Discussion by FredericHeem** [vanjs-org/van#34] [Ideas]
At the moment the bind function takes states and a function as parameters: 
`bind(state1, state2, (state1, state2)=> {//return a dom})`
It would be better for bind to have only 2 parameters: an array of state and the function:
`bind([state1, state2], (state1, state2) => {//return a dom})`

The bind implementation would be smaller and faster by removing the splice function.
Moreover the typescript definition would be much easier.
The only drawback is that it requires a migration of existing code.
What's your opinion on that subject?

**Comment by Tao-VanJS** [comment]
Thanks for the question! This is a design choice we made in the early days of VanJS. Basically we want to make the interface easier to use, as it aligns with VanJS's goal to reduce the entry barrier of UI programming. Regarding the overhead of `splice`, it can be optimized in future release by changing it to `pop`, thanks to the great [advice](https://github.com/vanjs-org/van/commit/3f98060971a8ff141179ef0fcd9f1b13b536a6f2#commitcomment-115001967) by @icecream17.
